syntax = "proto3";
package snailp;

message TransferRequest {
	bytes seq = 1;
	sfixed64 created = 2;

	message Fragment {
		bytes name = 1;
		fixed64 offset = 2;
		bytes ptr = 3;
		fixed32 crc32 = 4;
		bool eof = 5;
		fixed64 plain = 6;
		bytes uuid = 7;
	}
	repeated Fragment fragment = 3;
}
message TransferRespond {
	bytes seq = 1;
	sfixed32 tid = 2;
	sfixed64 created = 3;
	sfixed64 errnum = 4;
	bytes errstring = 5;
}

